app/src/main/java/com/example/themealsapp/data/repository/MealRepositoryImpl.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.themealsapp.data.repository
+
+import android.util.Log
+import com.example.themealsapp.data.local.MealDao
+import com.example.themealsapp.data.remote.MealsAPI
+import com.example.themealsapp.domain.model.Meal
+import com.example.themealsapp.domain.repository.MealRepository
+import com.example.themealsapp.utils.FailureResponse
+import com.example.themealsapp.utils.NullResponse
+import com.example.themealsapp.utils.UIState
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import javax.inject.Inject
+
+private const val TAG = "MealRepositoryImpl"
+class MealRepositoryImpl @Inject constructor(
+    private val mealsAPI: MealsAPI,
+    private val mealDao: MealDao
+) : MealRepository {
+
+    override fun getMealInfos(query: String): Flow<UIState<List<Meal>>> = flow {
+        emit(UIState.LOADING())
+
+        try {
+            Log.d(TAG, "getMealInfos: Fetching data from API")
+            val response = mealsAPI.searchMealByName(query)
+            if (response.isSuccessful) {
+                response.body()?.let { mealsResponse ->
+                    val mealInfos = mealsResponse.meals
+                    mealDao.deleteMeals(mealInfos?.map { it.strMeal })
```

Comment:
No need if you have conflict strategy as REPLACE

Reply:
You are right. Just Changed it